src/Rules/Generics/VarianceCheck.php

```diff
@@ -86,6 +87,41 @@ public function checkParametersAcceptor(
 		return $errors;
 	}
 
+	/** @return RuleError[] */
+	public function checkProperty(
+		PropertyReflection $propertyReflection,
+		string $message,
+		bool $isReadOnly,
+	): array
+	{
+		$readableType = $propertyReflection->getReadableType();
+		$writableType = $propertyReflection->getWritableType();
+
+		if ($readableType->equals($writableType)) {
+			$variance = $isReadOnly
+				? TemplateTypeVariance::createCovariant()
+				: TemplateTypeVariance::createInvariant();
+
+			return $this->check($variance, $readableType, $message);
+		}
+
+		$errors = [];
+
+		if ($propertyReflection->isReadable()) {
```

Comment:
Honestly, I'm not entirely sure if this part is correct. Could you give me an example of a property whose readable and writable types differ?

Reply:
Native property will always have the same read/write type.

Reply:
You can probably simplify this code by typehinting PhpPropertyReflection only (it's the return type of ClassReflection::getNativeProperty). And remove things that aren't possible (and thus untestable) with native properties.